realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -1315,7 +1337,8 @@ void BaseRealSenseNode::frame_callback(rs2::frame frame)
             {
                 auto f = (*it);
                 auto stream_type = f.get_profile().stream_type();
-                auto stream_index = f.get_profile().stream_index();
+                // auto stream_index = f.get_profile().stream_index();
+                auto stream_index = (stream_type==RS2_STREAM_INFRARED)?1:f.get_profile().stream_index();
```

Comment:
Why is this required? What if the stream type isn't INFRARED?

Reply:
[answered here](https://github.com/plusone-robotics/realsense/pull/21#discussion_r271960653)